service/src/test/java/uk/gov/hmcts/reform/fpl/service/ConfidentialDetailsServiceTest.java

```diff
@@ -1,130 +1,558 @@
+                    .telephoneNumber(Telephone.builder().telephoneNumber("01227 831393").build())
                     .build())
-                .build())
-            .build());
+                .build());
+        }
     }
+    
+    @Nested
+    class OthersTests {
+        @Test
+        void shouldAddToListWhenConfidentialDetailsForOther() {
+            List<Element<Other>> others = List.of(otherWithConfidentialFields(ID, CONFIDENTIAL));
 
-    private Element<Respondent> respondentElement(String hidden) {
-        return ElementUtils.element(Respondent.builder()
-            .party(RespondentParty.builder()
-                .firstName("James")
-                .contactDetailsHidden(hidden)
-                .address(Address.builder()
-                    .addressLine1("James' House")
+            List<Element<Other>> confidentialOthers = service.getConfidentialDetails(others);
+
+            assertThat(confidentialOthers).containsOnly(otherWithConfidentialFields(ID, NO_VALUE));
+        }
+
+        @Test
+        void shouldNotAddToListWhenOthersIsEmptyList() {
+            List<Element<Other>> confidentialOthers = service.getConfidentialDetails(List.of());
+
+            assertThat(confidentialOthers).isEmpty();
+        }
+
+        @Test
+        void shouldOnlyAddConfidentialDetailsToListOnlyWhenAnsweredYesForOther() {
+            List<Element<Other>> others = List.of(
+                otherWithConfidentialFields(ID, CONFIDENTIAL),
+                otherWithConfidentialFields(ID, NOT_CONFIDENTIAL));
+
+            List<Element<Other>> confidentialOthers = service.getConfidentialDetails(others);
+
+            assertThat(confidentialOthers).containsExactly(otherWithConfidentialFields(ID, NO_VALUE));
+        }
+
+        @Test
+        void shouldRemoveConfidentialDetailsWhenMarkedAsConfidential() {
+            List<Element<Other>> others = List.of(otherWithConfidentialFields(ID, CONFIDENTIAL));
+
+            assertThat(service.removeConfidentialDetails(others))
+                .containsOnly(otherWithRemovedConfidentialFields(ID));
+        }
+
+        @Test
+        void shouldNotRemoveConfidentialDetailsWhenNotMarkedAsConfidential() {
+            List<Element<Other>> others = List.of(otherWithConfidentialFields(ID, NOT_CONFIDENTIAL));
+
+            assertThat(service.removeConfidentialDetails(others)).isEqualTo(others);
+        }
+
+        @Test
+        void shouldReturnOthersIfOthersIsPrePopulated() {
+            CaseData caseData = CaseData.builder()
+                .others(Others.builder().firstOther(otherWithRemovedConfidentialFields(ID).getValue()).build())
+                .build();
+
+            List<Element<Other>> others = service.combineOtherDetails(caseData.getAllOthers(),
+                caseData.getConfidentialOthers());
+
+            assertThat(others.get(0).getValue()).isEqualTo(otherWithRemovedConfidentialFields(ID).getValue());
```

Comment:
a couple of places throughout we can change to unwrapElements to remove .get(0).getValue - I will leave up to you if you want to make this change or not

Reply:
resolving as we discussed